content/master/concepts/environment-configs.md

```diff
@@ -0,0 +1,291 @@
+            type: Value
+            value: my-label-value
+          - key: my-label-key
+            type: FromCompositeFieldPath
+            valueFromFieldPath: spec.parameters.deploy
+  resources:
+  # Removed for brevity
+```
+
+#### Manage selector results
+
+Selecting environments by labels may return more than one environment.  
+The Composition sorts all the results by the name of the environments and
+only uses the first environment in the sorted list. 
+
+Set the {{<hover label="selectResults" line="10">}}mode{{</hover>}} as 
+{{<hover label="selectResults" line="10">}}mode: Multiple{{</hover>}} to return
+all matched environments. Use 
+{{<hover label="selectResults" line="19">}}mode: Single{{</hover>}} to 
+return a single environment.
+
+{{<hint "note" >}}
+Sorting and the selection 
+{{<hover label="selectResults" line="10">}}mode{{</hover>}}
+only applies to a single 
+{{<hover label="selectResults" line="8">}}type: Selector{{</hover>}}. 
+
+This doesn't change how Compositions merge multiple
+{{<hover label="selectResults" line="7">}}environmentConfigs{{</hover>}}.
+{{< /hint >}}
+
+
+```yaml {label="selectResults"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example-composition
+spec:
+  environment:
+    environmentConfigs:
+    - type: Selector
+      selector:
+        mode: Multiple
+        matchLabels:
+          - key: my-label-key
+            type: Value
+            value: my-label-value
+          - key: my-label-key
+            type: FromCompositeFieldPath
+            valueFromFieldPath: spec.parameters.deploy
+    - type: Selector
+      selector:
+        mode: Single
+        matchLabels:
+          - key: my-other-label-key
+            type: Value
+            value: my-other-label-value
+          - key: my-other-label-key
+            type: FromCompositeFieldPath
+            valueFromFieldPath: spec.parameters.deploy
+```
+
+When using 
+{{<hover label="maxMatch" line="10">}}mode: Multiple{{</hover>}} limit the
+number of returned environments with
+{{<hover label="maxMatch" line="11">}}maxMatch{{</hover>}} and define the
+maximum number of environments returned. 
+
+The Composition sorts the returned environments alphabetically by name. Sort the
+environments on a different field with 
+{{<hover label="maxMatch" line="12">}}sortByFieldPath{{</hover>}} and define
+the field to sort by. 
+
+
+```yaml {label="maxMatch"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example-composition
+spec:
+  environment:
+    environmentConfigs:
+    - type: Selector
+      selector:
+        mode: Multiple
+        maxMatch: 4
+        sortByFieldPath: metadata.annotations[sort.by/weight]
+        matchLabels:
+          - key: my-label-key
+            type: Value
+            value: my-label-value
+          - key: my-label-key
+            type: FromCompositeFieldPath
+            valueFromFieldPath: spec.parameters.deploy
+```
+
```

Comment:
Do you think this would be a good place to reinforce for the reader again that now that multiple environments have been selected here, they will now all be merged with the top level list of environments?